handlers/update.go

```diff
@@ -161,6 +161,31 @@ func updateDeploymentSpec(
 
 		deployment.Spec.Template.Spec.Containers[0].LivenessProbe = probes.Liveness
 		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe = probes.Readiness
+
+		var annotations map[string]string
+		if request.Annotations != nil {
+			annotations = *request.Annotations
+		}
+
+		// if a policy is removed ... how do we remove the consequences?
```

Comment:
I would think - regenerate from the spec / factory?

Reply:
Not necessary as easy as it sounds, when there may be a default underlying value. 

Do we need a default policy so that we can revert fields to that?

Reply:
@alexellis i found the issue and it was actually a bug in my implementation and how i was comparing the old and new annotations. My last commit fixes the issue and adds a comment to highlight where the bug was, so that we can avoid it in the future